account-kit/signer/src/apiKeySigner.ts

```diff
@@ -0,0 +1,176 @@
+   *
+   * @param {Hex} msg The message to be signed
+   * @param {SignRawMessageMode} mode The signing mode (default is "ETHEREUM")
+   * @returns {Promise<Hex>} A promise that resolves to the signed message
+   */
+  public signRawMessage = async (
+    msg: Hex,
+    mode: SignRawMessageMode = "ETHEREUM"
+  ): Promise<Hex> => {
+    const user = await this.whoami();
+
+    const stampedRequest = await this.turnkeyClient.stampSignRawPayload(
+      buildStampedSignatureRequestBody(user, msg, mode)
+    );
+
+    const { signature } = await this.alchemyClient.signPayload(stampedRequest);
+
+    return signature;
+  };
+}
+
+/**
+ * AlchemyApiKeySigner is a SmartAccountSigner that uses an ApiKeySignerClient to sign messages.
+ * Primarily intended to be used server-side.
+ *
+ * @example
+ * ```ts
+ *  const signer = new AlchemyApiKeySigner({
+ *   apiKey: {
+ *     privateKey: "private-api-key",
+ *     publicKey: "public-api-key",
+ *   },
+ *   connection: {
+ *     apiKey: "alchemy-api-key",
+ *   },
+ *   orgId: "user-org-id",
+ * });
+ *
+ * const account = await createModularAccountV2({
+ *   transport,
+ *   signer,
+ *   chain,
+ * });
+ *
+ * const client = createAlchemySmartAccountClient({
+ *   account,
+ *   transport,
+ *   chain,
+ * });
+ * ```
+ */
+export class AlchemyApiKeySigner implements SmartAccountSigner {
+  signerType = "alchemy-api-key-signer";
+  inner: ApiKeySignerClient;
+
+  /**
+   * Constructs a new instance of the AlchemyApiKeySigner.
+   *
+   * @param {ApiKeySignerClientParams} params The parameters for the client, including the API key and orgId
+   */
+  constructor(params: ApiKeySignerClientParams) {
+    this.inner = new ApiKeySignerClient(params);
+  }
+
+  /**
+   * Retrieves the current user's address.
+   *
+   * @returns {Promise<Address>} A promise that resolves to the user's address
+   */
+  getAddress = async (): Promise<Address> => {
+    const { address } = await this.inner.whoami();
+    return address;
+  };
+
+  /**
+   * Signs a message using the inner client.
+   *
+   * @param {SignableMessage} msg The message to be signed
+   * @returns {Promise<Hex>} A promise that resolves to the signed message
+   */
+  signMessage = async (msg: SignableMessage): Promise<Hex> => {
+    const messageHash = hashMessage(msg);
+    return await this.inner.signRawMessage(messageHash);
+  };
+
+  /**
+   * Signs typed data using the inner client.
+   *
+   * @param {TypedDataDefinition<TTypedData, TPrimaryType>} params The typed data to be signed
+   * @returns {Promise<Hex>} A promise that resolves to the signed typed data
+   */
+  signTypedData = async <
+    const TTypedData extends TypedData | Record<string, unknown>,
+    TPrimaryType extends keyof TTypedData | "EIP712Domain" = keyof TTypedData
+  >(
+    params: TypedDataDefinition<TTypedData, TPrimaryType>
+  ): Promise<Hex> => {
+    const messageHash = hashTypedData(params);
+    return this.inner.signRawMessage(messageHash);
+  };
```

Comment:
Could you also implement `signAuthorization` here? It's optional in the interface, and would allow us to use this key w/ 7702.

Side note: it would be nice to create a utility that goes from a "raw signer" to a `SmartAccountSigner`, since this is logically just a copy of `AlchemyWebSigner` and much of `LocalAccountSigner`.

Reply:
will implement that

let me dig into that a bit more

Reply:
i see what you mean now, i am adding that

Reply:
hmmm the `LocalAccountSigner` doesn't actually use a raw signer though, as the local account actually implements a lot.